### ISSUE TITLE
Fault Tree Analysis has an infinite loop / gives an unhandled event loop exception

### ISSUE THREAD

**Issue by sprocter** [osate/osate2#2114]
## Summary

I'm trying to compute a fault tree for part of an aircraft's wheel braking system, but getting an error that there's an infinite loop / stack overflow occurring.

## Expected and Current Behavior

**Expected:** OSATE should generate and display a fault tree.

**Actual:** A popup box titled "Problem Occurred" appears, and says: 

> Unhandled event loop exception
> java.lang.StackOverflowError

The terminal fills up with these lines:

```
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:641)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.traverseCompositeErrorState(PropagationGraphBackwardTraversal.java:1011)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:714)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:641)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:641)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.traverseCompositeErrorState(PropagationGraphBackwardTraversal.java:1011)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:714)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:641)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:641)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.traverseCompositeErrorState(PropagationGraphBackwardTraversal.java:1011)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:714)
	at org.osate.aadl2.errormodel.PropagationGraph.util.PropagationGraphBackwardTraversal.processCondition(PropagationGraphBackwardTraversal.java:641)
```

## Steps to Reproduce

1. Load the "GTSE-Examples" project from the [GATSE repo](https://github.com/osate/osate2-gtse/tree/master/GTSE-Examples/ARP4761) into OSATE
2. Instantiate the `DecelerateAircraft.basic` system implementation, which is located in `GTSE-Examples/ARP4761/advanced/functional/decelerate.aadl`
3. Run the Fault Tree Analysis tool on the model with the following (default) settings:
![FTA Exception Screenshot](https://user-images.githubusercontent.com/1319948/70920897-4d23e980-1ff1-11ea-90d9-8e1eb8946f17.png)

## Environment

* **OSATE Version**: OSATE is built from source. I'm on the master branch; the most recent commit is fafd82c38837a4749b778656746156e3a390c4ab.
* **Operating System**: Mac OS 10.14.6

**Comment by jjhugues** [comment]
Seems also present in OSATE2.6.0

**Comment by lwrage** [comment]
Exception happens in 2.5.1, already, so this is not a newly introduced bug.

**Comment by reteprelief** [comment]
Addressed by fix for issue #2123 in pull request #2135
